### PR TITLE
Make tapThreshold a user settable property in component Canvas.java

### LINKED ISSUE #2366: Dragging on the canvas
**Describe the desired feature**

Could someone take a look at the canvas component? The idea is to modify the component so that some event returns all the coordinates from touching the canvas to moving to releasing. There is a "when.CanvasDraged" event but it shows the position only after 15 pixels. Link to the topic on the forum:

https://community.appinventor.mit.edu/t/asking-for-the-canvas-dragged-event/24264

### PR DESCRIPTION
**Pull request by Himanshu2107**
Closes #2366
Forum link : https://community.appinventor.mit.edu/t/asking-for-the-canvas-dragged-event/24264

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@Himanshu2107 Since you're updating an API related to a component, you will need to update the component's version number in YaVersion.java and specify (no-op) upgrades in YoungAndroidFormUpgrader.java (under appengine) and versioining.js (under blocklyeditor).

**Comment by Himanshu2107** [comment]
Changes addded. Anything else required?

**Review by ewpatton — CHANGES_REQUESTED** [review]
@Himanshu2107 A few things to address. First, you should reset YOUNG_ANDROID_VERSION as we will bump that just before merging. Second, merge `ucr` to pick up the latest changes. Lastly, there are some code style errors. After merging, run `ant checkstyle -Dcommit=ucr` to see the issues you need to address.

**Comment by Himanshu2107** [comment]
What do you mean by reset YOUNG_ANDROID_VERSION? Should I not increment the version myself or should I just incorporate the latest state of ucr and then increment the version?

**Comment by ewpatton** [comment]
@Himanshu2107 Yes, that's correct.  `YOUNG_ANDROID_VERSION` is meant to be monotonically increasing as it's needed to determine when upgrades should be applied. Since we don't know what order PRs will be merged, it's easier for the maintainers to update it once the PR is approved.

**Comment by Himanshu2107** [comment]
`ant checkstyle Dcommit=ucr' showed an error when executed. I tried `ant checkstyle Dcommit=upstream/ucr` which worked and spit out a bunch of checkstyle errors that I have fixed. Latest state of upstream/ucr is incorporated and I have left out the lines concerning `YOUNG_ANDROID_VERSION`.

File: appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java
Comment (aimiebell) [inline appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java:208]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

Comment (Himanshu2107) [reply appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java:208]: I think appinventor uses a different naming convention pertaining to methods concerning properties. The method name hence must start with an uppercase letter. The method name is reflected in the designer and should be the name of the property.

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

**Review by ewpatton — APPROVED** [review]
LGTM

**Pull request merged**
